sqlmesh/utils/transactional_file.py

```diff
@@ -26,48 +26,37 @@ def __init__(self, path: str, fs: fsspec.AbstractFileSystem) -> None:
         """Creates a new FileTransactionHandler.
 
         Args:
-            path: The path to lock. It should follow the format of a table name.
-                Something like snapshots/version=<ver>/<snapshot_id>
+            path: The path to lock.
+            fs: The fsspec file system.
         """
         self.path = path
+        self.lock_prefix = f"{self.path}.lock"
+        self.lock_path = f"{self.lock_prefix}.{time_uuid()}"
         self._fs = fs
         self._original_contents: t.Optional[bytes] = None
         self._is_locked = False
-        self._lock_id = uuid4().hex + str(get_ident())
-
-    @property
-    def lock_prefix(self) -> str:
-        """The prefix for the lock file."""
-        return f"{self.path}.lock."
 
-    @property
-    def lock_path(self) -> str:
-        """The path to the lock file."""
-        return self.lock_prefix + str(self._lock_id)
-
-    def _sync_locks(self) -> t.Dict[str, datetime]:
+    def _sync_locks(self) -> t.Dict[str, t.Tuple[float, str]]:
         """Gets a map of lock paths to their last modified times and removes stale locks."""
         output = {}
-        for lock in (
-            lock_info
-            for lock_info in self._fs.glob(
-                self.lock_prefix + "*", refresh=True, detail=True
-            ).values()
-            if os.path.basename(lock_info["name"]).startswith(os.path.basename(self.lock_prefix))
-        ):
+        now = datetime.now()
+
+        for lock in self._fs.glob(self.lock_prefix + ".*", refresh=True, detail=True).values():
```

Comment:
why `detail=True`?

Reply:
i have no idea, this was @z3z1ma's code

Reply:
https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=glob#fsspec.spec.AbstractFileSystem.ls i think you need detail for the timestamps